adafruit_max31856.py

```diff
@@ -0,0 +1,301 @@
+
+
+class MAX31856:
+    """Driver for the MAX31856 Universal Thermocouple Amplifier
+
+      :param ~busio.SPI spi_bus: The SPI bus the MAX31856 is connected to.
+      :param ~microcontroller.Pin cs: The pin used for the CS signal.
+      :param ~adafruit_max31856.ThermocoupleType thermocouple_type: The type of thermocouple.\
+      Default is Type K.
+
+    """
+
+    # A class level buffer to reduce allocations for reading and writing.
+    # Tony says this isn't re-entrant or thread safe!
+    _BUFFER = bytearray(3)
+
+    def __init__(self, spi, cs, thermocouple_type=ThermocoupleType.K):
+        self._device = SPIDevice(spi, cs, baudrate=500000, polarity=0, phase=1)
+
+        # assert on any fault
+        self._write_u8(_MAX31856_MASK_REG, 0x0)
+        # enable open circuit faults
+        self._write_u8(_MAX31856_CR0_REG, _MAX31856_CR0_OCFAULT0)
+
+        # set thermocouple type
+        # get current value of CR1 Reg
+        conf_reg_1 = self._read_u8(_MAX31856_CR1_REG)
+        conf_reg_1 &= 0xF0  # mask off bottom 4 bits
+        # add the new value for the TC type
+        conf_reg_1 |= int(thermocouple_type) & 0x0F
+        self._write_u8(_MAX31856_CR1_REG, conf_reg_1)
+
+    @property
+    def temperature(self):
+        """The temperature of the sensor and return its value in degrees celsius. (read-only)"""
+        self._perform_one_shot_measurement()
+
+        # read the 3-byte temperature
+        raw_read = self._read_u24(_MAX31856_LTCBH_REG)
+
+        # convert from 2's complement to negative int if sign bit is set
+        raw_read = raw_read - 0x1000000 if raw_read & (0x800000) else raw_read
+
+        # effectively shift raw_read >> 12 but retain significant figures
+        temp_float = (raw_read / 4096.0)
+
+        return temp_float
+
+    @property
+    def reference_temperature(self):
+        """The temperature of the cold junction in degrees celsius. (read-only)"""
+        self._perform_one_shot_measurement()
+        raw_read = self._read_u16(_MAX31856_CJTH_REG)
+
+        # if the sign bit(0x10000) is set, subtract 2^16 to get the negative value
+        raw_read = raw_read - 0x10000 if raw_read & (0x8000) else raw_read
+
+        # effectively shift raw_read >> 8 but retain significant figures
+        cold_junction_temp = raw_read / 256.0
+
+        return cold_junction_temp
+
+    @property
+    def temperature_thresholds(self):
+        """The thermocouple's low and high temperature thresholds
+        as a ``(low_temp, high_temp)`` tuple
+        """
+
+        low_temp_high_byte = self._read_u8(_MAX31856_LTLFTH_REG)
+        low_temp_low_byte = self._read_u8(_MAX31856_LTLFTL_REG)
+        high_temp_high_byte = self._read_u8(_MAX31856_LTHFTH_REG)
+        high_temp_low_byte = self._read_u8(_MAX31856_LTHFTL_REG)
+
+        low = round(((low_temp_high_byte << 8) | low_temp_low_byte)/16, 1)
+        high = round(((high_temp_high_byte << 8) | high_temp_low_byte)/16, 1)
+
+        return (low, high)
```

Comment:
This is due to the same thing you were dealing with for temperatures.
```python
>>> tc.temperature_thresholds = (20, -20)
>>> tc.temperature_thresholds
(20.0, 4076.0)
```
You may have to do some extra work in both the getter and setter methods to deal with this.